lib/bolt/cli.rb

```diff
@@ -379,710 +309,563 @@ def validate(options)
-                           config.default_inventoryfile
-                         end
+          # TODO: Configure logger with a separate method call?
+          Bolt::Logger.configure(config.log, config.color, config.disable_warnings)
+          Bolt::Logger.stream = config.stream
+          Bolt::Logger.analytics = analytics
+          Bolt::Logger.flush_queue
+
+          executor = Bolt::Executor.new(
+            config.concurrency,
+            analytics,
+            options[:noop],
+            config.modified_concurrency,
+            config.future
+          )
+
+          pal = Bolt::PAL.new(
+            Bolt::Config::Modulepath.new(config.modulepath),
+            config.hiera_config,
+            config.project.resource_types,
+            config.compile_concurrency,
+            config.trusted_external,
+            config.apply_settings,
+            config.project
+          )
+
+          plugins = Bolt::Plugin.setup(config, pal, analytics)
+
+          inventory = Bolt::Inventory.from_config(config, plugins)
+
+          log_outputter = Bolt::Outputter::Logger.new(options[:verbose], config.trace)
+
+          #
+          # FINALIZING SETUP
+          #
+
+          check_gem_install
+
+          # TODO: Discuss possibility of Bolt 4.0 having CLI overrides take
+          #       highest priority. Based on user expectations and discussion
+          #       in community, Bolt may be at a point where it no longer makes
+          #       sense for them to have lower priority than inventory.
+          warn_inventory_overrides_cli(config, options)
+
+          # TODO: See if this can be a method on Bolt::Analytics.
+          submit_screen_view(analytics, config, inventory, options)
+
+          process_target_list(plugins.puppetdb_client, @rerun, options)
+
+          # TODO: This is only needed because of casing issues in Puppet for Windows
+          #       paths. We should investigate finding and using the 'correct' casing
+          #       automatically so we don't need to warn users.
+          # Logger must be configured before checking path case and project file,
+          # otherwise logs will not display
+          config.check_path_case('modulepath', config.modulepath)
+
+          if options[:clear_cache] && File.exist?(config.project.plugin_cache_file)
+            FileUtils.rm(config.project.plugin_cache_file)
+          end
 
-      inventory_cli_opts = %i[authentication escalation transports].each_with_object([]) do |key, acc|
-        acc.concat(Bolt::BoltOptionParser::OPTIONS[key])
-      end
+          # TODO: Include logic for handling special cases when subscribing the
+          #       outputter (e.g. `plan run` and `apply`).
+          executor.subscribe(outputter)
+          executor.subscribe(log_outputter)
 
-      inventory_cli_opts.concat(%w[no-host-key-check no-ssl no-ssl-verify no-tty])
+          # TODO: Figure out where this should really go. It doesn't seem to
```

Comment:
Yeah, this and the clear cache lines above + script line below don't really have a clear organizational place. I think you could group them in a separate function call if you want, or they're fine here :shrug: I personally don't mind long functions especially since a lot of this is a single function call with newlined arguments.